### PR TITLE
Add caseID, executionCondition, and cancellationCondition parameters

### PR DESCRIPTION
**Pull request by mwshortt**
Add additional optional parameters to the sendPayment and executePayment functions
Update expected format for caseID parameter to: notary/cases/uuid

### CODE REVIEW

File: src/notaryUtils.js [hunk 1]
```diff
@@ -26,11 +26,15 @@ function createCaseID () {
  */
 function setupCase (params) {
   return co(function * () {
-    const uniqueID = params.caseID || uuid()
+    // the caseID param will be in the format of notary/cases/uuid,
+    // only check if its uuid is less than 40 characters
+    const uniqueID = params.caseID ? params.caseID.substring(params.caseID.indexOf('/cases/') + 7) : uuid()
```
Comment (emschwartz) [inline src/notaryUtils.js]: What is the `+ 7` for?

Comment (emschwartz) [inline src/notaryUtils.js]: Why bother passing in the `caseID` as a URI if you're going to throw away most of it and just keep the uuid?

Comment (mwshortt) [inline src/notaryUtils.js]: We are only "throwing" it away for the uniqueID check. If you see on line 37, we are using the URI in its entirety if provided

Comment (emschwartz) [inline src/notaryUtils.js]: Hmm, it seems a little bit odd that the `caseID` is the only parameter that gets validated at all. Is there a specific reason to have that check that the unique id is the right length?

Comment (mwshortt) [inline src/notaryUtils.js]: The length is a database restriction, see @lumberj's comment on a previous PR: https://github.com/interledger/five-bells-sender/pull/47#discussion_r56205805. Though that may be for the entire `caseID` URI not the "uniqueID", I am not sure. @lumberj could you provide some context?

Comment (emschwartz) [inline src/notaryUtils.js]: If the id provided doesn't match the `Uuid` schema the notary will return an error when you try to set up the case (it's checked [here](https://github.com/interledger/five-bells-notary/blob/master/src/controllers/cases.js#L22)). I think the sender should either 1) not check any of the parameters and just return any errors returned by the components or 2) check all of the parameters it's given

Comment (justmoon) [inline src/notaryUtils.js]: What if you use a notary that uses a differently formatted case ID? In REST you should not make any assumptions about what URIs look like internally.

It's true that f-b-sender already makes assumptions about the shape of the URI, but that's precisely why allowing you pass in a custom URI is a good idea. But if you let the user pass in a URI, you should treat it as an opaque string, not try to parse/validate it.

As an example, here is a perfectly valid case URI:

https://bob-notary.com/ilp-notary/123

**Pull request closed without merging**